lib/src/equality.dart

```diff
@@ -272,7 +272,7 @@ class UnorderedIterableEquality<E> extends _UnorderedEquality<E, Iterable<E>?> {
 /// The [equals] and [hash] methods accepts `null` values,
 /// even if the [isValidKey] returns `false` for `null`.
 /// The [hash] of `null` is `null.hashCode`.
-class SetEquality<E> extends _UnorderedEquality<E, Set<E>?> {
+class SetEquality<E> extends _UnorderedEquality<E, Set<E>> {
```

Comment:
This made `SetEquality<int>` implement `Equality<Set<int>?>`, which means that writing:
```dart
Equality<Set<X>> eq = SetEquality<X>(xEquality);
```
would be a type error. Seems nobody hit that yet, but fixing it to be certain.

(We still let `SetEquality<X>` accept `Set<X>?` as arguments to equality, but that's just being permissive in what it accepts, not in what it *is*).

Reply:
Could we handle this change in a separate PR?